### ISSUE TITLE
Excessive battery consumption (with push notifications enabled)

### ISSUE THREAD

**Issue by stigger** [ChatSecure/ChatSecure-iOS#553]
Not sure what causes it, there were no more than 10-15 messages received via push notifications.
![img_0017](https://cloud.githubusercontent.com/assets/769359/17833351/147e39d8-671b-11e6-82d2-0fee412ef719.jpg)

**Comment by chrisballinger** [comment]
The app will wake up for ~30 seconds per push notification, and still stay
open for ~2 minutes after opening the app normally. Is your server using
XEP-0357?

**Comment by stigger** [comment]
Yes, I know. Still, the consumption is way too high.

Yes, server supports XEP-0357 and sends push notifications.

**Comment by chrisballinger** [comment]
What is your server configuration?

I'm wondering if extraneous background pushes are being sent from your server for messages without a body, like typing notifications. Can you check your server logs to see how often push events are being sent?

**Comment by stigger** [comment]
The server is this: https://github.com/weiss/ejabberd/commits/push

I additionally tweaked the mod_push to only send pushes if there is a message and ignore the rest.

**Comment by chrisballinger** [comment]
Excessive 0357 pushes to our server is the only thing I can think of at the moment, because we can't keep the app open that long otherwise. I've only tested with mod_cloud_notify and Prosody, I'm not very familiar with ejabberd/mod_push. I think this might be a larger issue for others because our app server saw a HUGE increase in the number of pushes when we turned on 0357, way more than I expected.
